polkadot-parachain/tests/purge_chain_works.rs

```diff
@@ -15,59 +15,39 @@
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 use assert_cmd::cargo::cargo_bin;
+use nix::sys::signal::{SIGINT, SIGTERM};
 use std::{convert::TryInto, process::Command, thread, time::Duration};
+use tempfile::tempdir;
 
 mod common;
 
-#[test]
+#[tokio::test]
 #[cfg(unix)]
 #[ignore]
-fn purge_chain_works() {
-	fn run_node_and_stop() -> tempfile::TempDir {
-		use nix::{
-			sys::signal::{kill, Signal::SIGINT},
-			unistd::Pid,
-		};
-
-		let base_path = tempfile::tempdir().unwrap();
-
-		let mut cmd = Command::new(cargo_bin("polkadot-parachain"))
-			.args(&["-d"])
-			.arg(base_path.path())
-			.args(&["--", "--dev"])
-			.spawn()
-			.unwrap();
-
-		// Let it produce some blocks.
-		thread::sleep(Duration::from_secs(30));
-		assert!(cmd.try_wait().unwrap().is_none(), "the process should still be running");
+async fn purge_chain_works() {
+	// Check that both databases are deleted
 
-		// Stop the process
-		kill(Pid::from_raw(cmd.id().try_into().unwrap()), SIGINT).unwrap();
-		assert!(common::wait_for(&mut cmd, 30).map(|x| x.success()).unwrap_or_default());
+	let base_dir = tempdir().expect("could not create a temp dir");
 
-		base_path
-	}
+	let args = &["--", "--dev"];
 
-	// Check that both databases are deleted
-	{
-		let base_path = run_node_and_stop();
+	common::run_node_for_a_while(base_dir.path(), args, SIGINT).await;
+	common::run_node_for_a_while(base_dir.path(), args, SIGTERM).await;
 
-		assert!(base_path.path().join("chains/local_testnet/db/full").exists());
-		assert!(base_path.path().join("polkadot/chains/dev/db/full").exists());
+	assert!(base_dir.path().join("chains/local_testnet/db/full").exists());
+	assert!(base_dir.path().join("polkadot/chains/dev/db/full").exists());
 
-		let status = Command::new(cargo_bin("polkadot-parachain"))
-			.args(&["purge-chain", "-d"])
-			.arg(base_path.path())
-			.arg("-y")
-			.status()
-			.unwrap();
-		assert!(status.success());
+	let status = Command::new(cargo_bin("polkadot-parachain"))
+		.args(&["purge-chain", "-d"])
+		.arg(base_dir.path())
+		.arg("-y")
+		.status()
+		.unwrap();
+	assert!(status.success());
 
-		// Make sure that the `parachain_local_testnet` chain folder exists, but the `db` is deleted.
-		assert!(base_path.path().join("chains/local_testnet").exists());
-		assert!(!base_path.path().join("chains/local_testnet/db/full").exists());
-		// assert!(base_path.path().join("polkadot/chains/dev").exists());
```

Comment:
@ruseinov 
I would maybe remove those lines, if not needed anymore

Reply:
I do not want to change what was already there, simply adopting a new method of running the node. I think we should leave cleanups for any follow up PRs as I don't have any context as to why those were left commented out.